frameworks/CSharp/FastHttpApi/PlatformBenchmarks/Program.cs

```diff
@@ -0,0 +1,137 @@
+        {
+            if (e.Session == null)
+            {
+                server.Log(LogType.Error, null, "server error {0}@{1}\r\n{2}", e.Message, e.Error.Message, e.Error.StackTrace);
+            }
+            else
+            {
+                server.Log(LogType.Error, null, "session {2}@{3} error {0}@{1}\r\n{4}", e.Message, e.Error.Message, e.Session.RemoteEndPoint, e.Session.ID, e.Error.StackTrace);
+            }
+        }
+
+        public virtual void Log(IServer server, ServerLogEventArgs e)
+        {
+            Console.WriteLine("[{0}:{2}] {1}", e.Type, e.Message, DateTime.Now);
+        }
+
+        public virtual void SessionDetection(IServer server, SessionDetectionEventArgs e)
+        {
+            ;
+        }
+
+        public virtual void SessionPacketDecodeCompleted(IServer server, PacketDecodeCompletedEventArgs e)
+        {
+
+        }
+
+        private void Onplaintext(PipeStream stream)
+        {
+            HttpParse.Plaintext(stream);
+        }
+
+        private void OnJson(PipeStream stream)
+        {
+            HttpParse.Json(stream);
+        }
+
+        public virtual void SessionReceive(IServer server, SessionReceiveEventArgs e)
+        {
+            PipeStream stream = e.Stream.ToPipeStream();
+            string line = null;
+            if (stream.TryReadWith(HttpParse.LineBytes, out line))
+            {
+                stream.ReadFree((int)stream.Length);
+                Tuple<string, string, string> request = HttpParse.AnalyzeRequestLine(line);
+                if (request.Item2 == "/plaintext")
+                {
+                    Onplaintext(stream);
+                }
+                else if (request.Item2 == "/json")
+                {
+                    OnJson(stream);
+                }
+                else if (request.Item2 == "/")
+                {
+                    stream.Write(HttpParse.BeetleXServerBytes, 0, HttpParse.BeetleXServerBytes.Length);
+                }
+                else
+                {
+                    stream.Write(HttpParse.NotFoundBytes, 0, HttpParse.NotFoundBytes.Length);
+                }
+                stream.Flush();
+            }
+            else
+            {
+                e.Session.Dispose();
+            }
+
+        }
+
+        private BeetleX.IServer mServer;
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            NetConfig config = new NetConfig();
+            config.Port = 8080;
+            config.LogLevel = LogType.Warring;
+            mServer = SocketFactory.CreateTcpServer(config, this, null);
+            mServer.Open();
+            Console.WriteLine("BeetleX base server");
+            Console.WriteLine($"ServerGC:{System.Runtime.GCSettings.IsServerGC}");
+            Console.Write(mServer);
+            return Task.CompletedTask;
+        }
+
+        public async Task StopAsync(CancellationToken cancellationToken)
+        {
+            return;
```

Comment:
You probably want to stop the server here

Reply:
Can I do this?
```
            var builder = new HostBuilder()
                .ConfigureServices((hostContext, services) =>
                {
                    services.AddHostedService<BeetleXHttpServer>();
                });
            var result = builder.Start();
```
```
        public virtual async Task StopAsync(CancellationToken cancellationToken)
        {
            mApiServer.BaseServer.Dispose();
            return;
        }
```